src/NameFieldSet.tsx

```diff
@@ -59,6 +59,32 @@ export class NamesFieldSet extends React.Component<INameFieldSetProps, INameFiel
     this.setState({ names: [...this.state.names, ""] });
   }
 
+  private arrayContainsDuplicates = (a: string[]) => {
+    const uniqueSet:Set<string> = new Set();
+    for(let i = 0; i <= a.length; i++) {
+      if (a[i] == null) {
+        continue
+      }
+      const value = a[i].toLowerCase();
+      if (uniqueSet.has(value)) {
+        return true;
+      }
+      uniqueSet.add(value)
+    }
+    return false;
+  }
+
+  private arrayContainsString = (a: string[], s: string) => {
```

Comment:
a and s are not descriptive names. Replace with nameArray and name, respectively, for instance.